config/ogdl/src/main/java/org/apache/shiro/config/ogdl/ReflectionBuilder.java

```diff
@@ -287,10 +287,15 @@ private String parseBeanId(String lhs) {
             }
 
             processor.execute();
-        }
 
-        //SHIRO-413: init method must be called for constructed objects that are Initializable
-        LifecycleUtils.init(objects.values());
+            //SHIRO-778: onInit method on AuthenticatingRealm is called twice
+            objects.keySet().stream()
+                    .filter(key -> !kvPairs.keySet().contains(key))
```

Comment:
I think it is not necessary to call `.keySet()` when you can just use `.containsKey(key)`. This will create a new view each time, the latter might also be a little bit easier to read.

Reply:
Right